### ISSUE TITLE
[BUG] COA Single Sign On Button Error in Knack Apps

### ISSUE THREAD

**Issue by ChristinaTremel** [cityofaustin/atd-data-tech#15132]
**Describe the bug**
- Ronnie called me on Teams and brought to my attention that the Single Sign On button in Knack is not allowing people to sign in. 
- If people are already logged into Knack, everything is working normally. To recreate the bug, I signed out and tried to sign back in using the SSO button and there's an error that refreshes to the same log in page.
- The URL while trying to sign in is: https://atd.knack.com/amd#undefined/?klem=An%20error%20occurred%20when%20attempting%20to%20authenticate%20with%20COACD.%20Please%20try%20again%20later
- I've tried logging into AMD Data Tracker, Finance, and the HR App and it's happening across the board.

**URL**
- https://atd.knack.com/signs-markings#home/
- https://atd.knack.com/amd#home/
- https://atd.knack.com/finance-purchasing#home/
- https://atd.knack.com/hr#home/

**What steps can we follow to reproduce the bug you are seeing?**
1. Go to Knack App login page
2. Click on big `Sign-On` button
3. See error

**Comment by dianamartin** [comment]
# 1/3/2024
- [x] Had Scott email out all users
>All,
All Knack applications are having an outage. The single sign on button is loading and not allowing users to sign into knack applications. The Data and Technology Services team is aware of the issue and the vendor is actively working to resolve it.
Work around: 
•	Click “Non-COA Sign-In” button
•	Click “forgot?” to reset your password
•	Then enter your email address into the “Email Address” field and click “Submit” 
o	Then an email will be sent with instructions on resetting your password
•	Use your reset password and the “Non-COA Sign-In” method to log into your knack applications.
NOTE: You’ll have to do this for each individual knack application you are trying to sign into 

- We provided screen shots to log in as a work around
- [x] created CTM Ticket `INC0354244`
- [x] Reached out to CTM (Michael S. and John D.) to try to resolve this issue quickly

**Comment by chiaberry** [comment]
I tested signing in from the admin center on the microsoft side, and I got this response, its a better error message that might be helpful to cite:

"https://atd.knack.com/amd#home/?klem=Invalid%20signature"

**Comment by dianamartin** [comment]
# 1/3/2024
In this slack [thread ](https://austininnovation.slack.com/archives/CCUPS4GBX/p1704304814788689) we were able to discover that the "Authentication Context" field was no longer needed. Chrispin had tested it on the TPW Forms app and I tested it on the ATD Visitor Sign in app and it worked. 
- He updated all the knack apps to remove this field and switch it to "None"
- I sent out and email to Scott/Adan, he'll probably send out the resolution email tomorrow
- [ ] Waiting on resolution email to be sent

**Comment by dianamartin** [comment]
# 1/4/24
Scott sent email out at 7:19 am
>All,
The single sign on issue in Knack applications has been resolved. All applications should be working normally now.
Let us know if you experience any further issues.
